postprocess.py

```diff
@@ -92,51 +94,61 @@ def parse_args():
+        for size in sizes:
+            size = size.strip().split("\t")
+            if size[0] in uniq_chroms:
+                sizetuple.append((size[0], int(size[1])))
+    bw = pyBigWig.open(bwoutputfilename, "w")
+    bw.addHeader(sizetuple)
+    bw.addEntries(list(batch_data["chrom"]), list(batch_data["start"]), ends=list(batch_data["end"]), values=list(batch_data["score"]))
+    bw.close()
 
 def writer(batch_range, outfilename):
     start, end = batch_range[0], batch_range[1]
-    # num_batches = end - start
-    with open(outfilename, 'w') as outfile:
-        with h5py.File(args.predictions_file, 'r') as infile:
-            # Load predictions
-            if args.channel is not None:
-                scores = infile['data'][start:end, :, args.channel]
-            else:
-                scores = np.array(infile['data'])
-
-            # Flatten scores
-            scores = scores.flatten()
-
-            # Threshold predictions - for classification output
-            if args.threshold is not None:
-                scores = (scores > args.threshold).astype(int)
-            # Round scores - for regression output
-            elif args.round is not None:
-                scores = scores.astype('float64')
-                # Sometimes np.around doesn't work with float32. To investigate.
-                scores = np.around(scores, decimals=args.round)
-
-            # if the batch contains values > 0, write them
-            if (scores > 0).any():
-                # Select intervals corresponding to batch
-                batch_intervals = intervals.iloc[start:end, :]
-                # Expand each interval and combine with scores
-                batch_bg = intervals_to_bg(batch_intervals, scores)
-
-                # Write to file
-                df_to_bedGraph(batch_bg, outfile)
+    with h5py.File(args.predictions_file, 'r') as infile:
+        # Load predictions
+        if args.channel is not None:
+            scores = infile['data'][start:end, :, args.channel]
+        else:
+            scores = np.array(infile['data'])
+
+        # Flatten scores
+        scores = scores.flatten()
+
+        # Threshold predictions - for classification output
+        if args.threshold is not None:
+            scores = (scores > args.threshold).astype(int).astype('float64')
+        # Round scores - for regression output
+        elif args.round is not None:
+            scores = scores.astype('float64')
+            # Sometimes np.around doesn't work with float32. To investigate.
+            scores = np.around(scores, decimals=args.round)
+
+        # if the batch contains values > 0, write them
+        if (scores > 0).any():
+            # Select intervals corresponding to batch
+            batch_intervals = intervals.iloc[start:end, :]
+            # Expand each interval and combine with scores
+            batch_bg = intervals_to_bg(batch_intervals, scores)
+            df_to_bigwig(batch_bg, outfilename)
```

Comment:
as discussed earlier, let's keep support for both `df_to_bigwig` and `df_to_bedgraph` depending on an input argument?

Reply:
Yes, I aim to do that as a separate PR if that's okay.